core/db/db.go

```diff
@@ -16,18 +19,19 @@ type InstanceID string
 
 // NewInstanceID generates a new identity for an instance.
 func NewInstanceID() InstanceID {
-	return InstanceID(uuid.New().String())
+	t := time.Now().UTC()
+	entropy := ulid.Monotonic(rand.New(rand.NewSource(t.UnixNano())), 0)
+	id := ulid.MustNew(ulid.Timestamp(t), entropy)
+	return InstanceID(id.String())
 }
 
 func (e InstanceID) String() string {
 	return string(e)
 }
 
+// IsValidInstanceID checks if an id is valid
 func IsValidInstanceID(instanceID string) bool {
-	// ToDo: Decide how we want to apply instance id requirements
 	return len(instanceID) > 0
-	// _, err := uuid.Parse(instanceID)
-	// return err == nil
```

Comment:
We can always add more to this in the future, but this is the only check needed for now since we're allowing clients to specify any string as an id.